### PR TITLE
ScriptBuilder: add ability to build scripts with `creationTime`

### PR DESCRIPTION
**Pull request by schildbach**

### CODE REVIEW

File: core/src/main/java/org/bitcoinj/script/ScriptBuilder.java [hunk 1]
```diff
@@ -457,14 +476,18 @@ public static Script createP2PKOutputScript(ECKey pubKey) {
...
-        builder.op(OP_EQUALVERIFY);
-        builder.op(OP_CHECKSIG);
-        return builder.build();
+        checkState(chunks.isEmpty());
+        op(OP_DUP);
+        op(OP_HASH160);
+        data(hash);
+        op(OP_EQUALVERIFY);
+        op(OP_CHECKSIG);
+        return this;
```
Comment (msgilligan) [inline core/src/main/java/org/bitcoinj/script/ScriptBuilder.java:495]: Can't this be:
```
    return this.op(OP_DUP)
                     .op(OP_HASH160)
                     ...
                     .op(OP_CHECKSIG);
```   
?

**Review by msgilligan — APPROVED** [review]
Looks good!  A few JavaDoc requests and a suggestion to keep fluent method calls using `this`.

File: core/src/main/java/org/bitcoinj/script/ScriptBuilder.java [hunk 2]
```diff
@@ -262,34 +272,48 @@ public ScriptBuilder opFalse(int index) {
...
         }
+        return this;
+    }
+
+    private ScriptBuilder p2whOutputScript(SegwitAddress address) {
+        checkState(chunks.isEmpty());
+        // OP_0 <pubKeyHash|scriptHash>
+        smallNum(address.getWitnessVersion());
+        data(address.getWitnessProgram());
+        return this;
```
Comment (msgilligan) [inline core/src/main/java/org/bitcoinj/script/ScriptBuilder.java:316]: I think you can use method-chaining here, too.

File: core/src/main/java/org/bitcoinj/script/ScriptBuilder.java [hunk 3]
```diff
@@ -262,34 +272,43 @@ public ScriptBuilder opFalse(int index) {
...
-        return new Script(chunks);
+        return new Script(chunks, creationTime);
     }
 
     /** Creates an empty script. */
     public static Script createEmpty() {
         return new ScriptBuilder().build();
     }
 
+    /** Creates a scriptPubKey that encodes payment to the given address. */
```
[the remark this answers is not in the document]
Comment (schildbach) [reply core/src/main/java/org/bitcoinj/script/ScriptBuilder.java:283]: Agreed.

File: core/src/main/java/org/bitcoinj/script/ScriptBuilder.java (hunk 1 above)
Comment (schildbach) [reply core/src/main/java/org/bitcoinj/script/ScriptBuilder.java:495]: Yes, ok, I  changed this for the methods I touched.

**Comment by schildbach** [comment]
Merged.

**Pull request closed without merging**